cluster/addons/metadata-proxy/gce/metadata-proxy-configmap.yaml

```diff
@@ -28,33 +28,33 @@ data:
 
         # Allow for REST discovery.
         location = / {
-            if ($args ~ "recursive=true") {
+            if ($args ~* "recursive") {
                     return 403 "?recursive calls are not allowed by the metadata proxy.";
             }
             proxy_pass http://169.254.169.254;
         }
         location = /computeMetadata/ {
-            if ($args ~ "recursive=true") {
+            if ($args ~* "recursive") {
```

Comment:
I'm now afraid this will prevent `?recursive=false`, too. @ihmccreery if you've figured out the values that count as `true`, I think we should add them here. (But then I'll probably worry about if this $args~* check actually normalizes the URLencoding.)

Reply:
I feel okay about blocking `?recursive=false` (and `?recursive=0`, etc.), since, as far as I can tell, there's no conventions on a strict set for what to accept as booleans.  Params are just a bunch of strings that can be interpreted however the server wants.

Reply:
(From some messing around, `true`, `1`, `yes`, and `y` are all accepted.)

Reply:
`on` is also accepted, tried because I found spring-framework's [StringToBooleanConverter.java](https://github.com/spring-projects/spring-framework/blob/master/spring-core/src/main/java/org/springframework/core/convert/support/StringToBooleanConverter.java).

Reply:
On a second thought maybe blocking `?recursive` is fine. I'm not familiar with the nginx.conf language enough to see if it warrants a pattern like `^recursive=` (maybe at least leave the `=` in for more precise matching).